pages/api/system/reload.js

```diff
@@ -195,24 +195,64 @@ export default async function handler(req, res) {
       }
 
       // - events
+      async function getCoordinates(city,state,country) {
+        const location = await fetch(
+          `https://nominatim.openstreetmap.org/?addressdetails=1&city=${city}&state=${state}&country=${country}&format=json&limit=1`
+        );
+        const coordinates = await location.json();
+        const final = {
+          lat: await coordinates[0].lat,
+          lon: await coordinates[0].lon,
+        };
+        return final;
+      }
       try {
         if (profile.events) {
           await Profile.findOneAndUpdate(
```

Comment:
After debugging, the following code handles our user input:
```js
const events = await Promise.all(
  profile.events.map(async (event, position) => {
    let location = {}
    if (new Date(event.date.start) < Date.now()) {
      return location;
    }
    if (event.location) {
      location = {
        location: {...event.location}
      }
      const coordinates = await getCoordinates(
        event.location.city,
        event.location.state,
        event.location.country
      );
      if (coordinates) {
        location.location.lat = coordinates.lat;
        location.location.lon = coordinates.lon;
      }
    }
    return {
      order: position,
      ...event,
      ...location
    };
}));

await Profile.findOneAndUpdate(
  { username: profile.username },
  { events }
);
```

Reply:
```
let location = {}
if (new Date(event.date.start) < Date.now()) {
 return location;
}
```
Here, what you are doing is, if `new Date(event.date.start) < Date.now()` this condition comes true, then you are returning blank value for event. But, in the database we have to store the values provided by the user right? 
 `new Date(event.date.start) < Date.now()` this condition must only be checked for fetching the coordinates.

```
const events = profile.events ? await Promise.all(
  profile.events.map(async (event, position) => {
    let location = {};
    if (event.location) {
      location = {
        location: { ...event.location },
      };
      if (new Date(event.date.start) > Date.now()) {
        const coordinates = await getCoordinates(
          event.location.city,
          event.location.state,
          event.location.country
        );
        if (coordinates) {
          location.location.lat = coordinates.lat;
          location.location.lon = coordinates.lon;
        }
      }
    }
    return {
      order: position,
      ...event,
      ...location,
    };
  })
): null;

await Profile.findOneAndUpdate(
  { username: profile.username },
  { events }
);
```
This can be the updated code where i have put `if (new Date(event.date.start) > Date.now()) {` this condition for fetching the coordinates.

Reply:
```js
    return {
      order: position,
      ...event,
      ...location,
    };
```
This code sets the order field, copies in all the fields from the event including location if present, then updates the location. If the location variable is `{}`, then any location fields in the event will retained as is.

Reply:
```
let location = {}
if (new Date(event.date.start) < Date.now()) {
  return location;
}
```
Here, you are directly returning a blank value.

Reply:
Yes, you are correct.
Without my colored bracket setup, I can get lost in the code at times.

FYI, if you put `js` after the three backticks to open a code block, it will colorize it as JavaScript.

Reply:
Ohh.. thank you. I was just about to ask you how your text is getting coloured. 
Will push my code now.